src/redux/categories/categories.js

```diff
@@ -0,0 +1,35 @@
+const ADDSTATUS = 'ADDSTATUS';
+const REMOVESTATUS = 'REMOVESTATUS';
+
+const initialState = [];
+
+export const addcategory = (category) => ({
+  type: ADDSTATUS,
+  category,
+});
+
+export const delcategory = (category) => ({
+  type: REMOVESTATUS,
+  category,
+});
+
+const categoryReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADDSTATUS: return [
+      ...state, action.payload,
+    ];
+
+    case REMOVESTATUS: return [
+      ...state.filter((each) => each.id !== action.payload.id),
+    ];
```

Comment:
- [ ] Kindly Define only CHECK_STATUS, you don't need to define add and delete categories as action types. The value of each action type needs to follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux), so it will be something like `const CHECK_STATUS = bookstore/categories/CHECK_STATUS`.
- [ ] Also define a case for `CHECK_STATUS` which return `'Under Construction'` and the default one return state.